File: example/src/main/java/com/example/bundlerexample/Example2Activity.java
package com.example.bundlerexample;

import android.os.Bundle;
import android.util.Log;

import com.example.bundlerexample.model.User;
import com.hendraanggrian.bundler.Bundler;
import com.hendraanggrian.bundler.annotations.BindExtra;
import com.hendraanggrian.bundler.annotations.WrapExtras;

@WrapExtras
public class Example2Activity extends BaseActivity {

    @BindExtra("user") User user;

    @Override
    protected int getContentLayout() {
        return R.layout.activity_example1;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundler.bind(this);
        Log.d("user.name", user.name);
    }
}

File: example/src/com/example/bundlerexample/Example2Activity.kt
package com.example.bundlerexample

import android.os.Bundle
import android.util.Log
import com.example.bundlerexample.model.User
import com.hendraanggrian.bundler.BindExtra
import com.hendraanggrian.bundler.Bundler

class Example2Activity(override val contentLayout: Int = R.layout.activity_example1) : BaseActivity() {

    @BindExtra("user") lateinit var user: User

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Bundler.bindExtras(this)
        Log.d("user.name", user.name)
    }
}